pkg/cmd/stack/deploy.go

```diff
@@ -698,19 +701,48 @@ func validateDefinedServices(s *model.Stack, servicesToDeploy []string) error {
 	return nil
 }
 
-func addDependentServicesIfNotPresent(ctx context.Context, s *model.Stack, options *StackDeployOptions, c kubernetes.Interface) {
-	added := make([]string, 0)
-	for _, svcToDeploy := range options.ServicesToDeploy {
+// AddDependentServicesIfNotPresent adds dependands services to deploy
+func AddDependentServicesIfNotPresent(ctx context.Context, s *model.Stack, svcsToDeploy []string, c kubernetes.Interface) []string {
+	initialSvcsToDeploy := svcsToDeploy
+	svcsToDeployWithDependencies := addDependentServices(ctx, s, svcsToDeploy, c)
+	if len(initialSvcsToDeploy) != len(svcsToDeploy) {
+		added := getAddedSvcs(initialSvcsToDeploy, svcsToDeployWithDependencies)
+
+		oktetoLog.Warning("The following services need to be deployed because the services passed as arguments depend on them: [%s]", strings.Join(added, ", "))
+	}
+	return svcsToDeployWithDependencies
+}
+
+func addDependentServices(ctx context.Context, s *model.Stack, svcsToDeploy []string, c kubernetes.Interface) []string {
+	initialLength := len(svcsToDeploy)
+	for _, svcToDeploy := range svcsToDeploy {
 		for dependentSvc := range s.Services[svcToDeploy].DependsOn {
-			if !isSvcToBeDeployed(options.ServicesToDeploy, dependentSvc) && !isSvcRunning(ctx, s.Services[dependentSvc], s.Namespace, dependentSvc, c) {
-				options.ServicesToDeploy = append(options.ServicesToDeploy, dependentSvc)
-				added = append(added, dependentSvc)
+			if !isSvcToBeDeployed(svcsToDeploy, dependentSvc) && !isSvcRunning(ctx, s.Services[dependentSvc], s.Namespace, dependentSvc, c) {
+				svcsToDeploy = append(svcsToDeploy, dependentSvc)
 			}
 		}
 	}
-	if len(added) > 0 {
-		oktetoLog.Warning("The following services need to be deployed because the services passed as arguments depend on them: [%s]", strings.Join(added, ", "))
+	if initialLength != len(svcsToDeploy) {
+		return addDependentServices(ctx, s, svcsToDeploy, c)
+	}
+	return svcsToDeploy
+}
+
+func getAddedSvcs(initialSvcsToDeploy, svcsToDeployWithDependencies []string) []string {
```

Comment:
We could reduce the reading (and computational) complexity of this if we transform the `initialSvcsToDeploy` into a set